crates/core/src/logstore/mod.rs

```diff
@@ -179,8 +240,24 @@ pub enum PeekCommit {
 pub struct LogStoreConfig {
     /// url corresponding to the storage location.
     pub location: Url,
-    /// Options used for configuring backend storage
-    pub options: StorageOptions,
+    // Options used for configuring backend storage
+    pub options: StorageConfig,
+}
+
+impl LogStoreConfig {
+    pub fn decorate_store<T: ObjectStore + Clone>(
+        &self,
+        store: T,
+        table_root: Option<&url::Url>,
+        handle: Option<Handle>,
+    ) -> DeltaResult<Box<dyn ObjectStore>> {
+        let table_url = table_root.unwrap_or(&self.location);
+        self.options.decorate_store(store, table_url, handle)
+    }
+
+    pub fn object_store_factory(&self) -> ObjectStoreFactoryRegistry {
+        self::factories::object_store_factories()
+    }
 }
```

Comment:
the logstore config is a good way for us to inject functionality into the logstore, since the API dictates log stores also expose it again, but we control the implementation... this should alos gove us a way to just inject a registry taken from - let's say datafusion session - and wire it through.